### PR TITLE
Add a first draft of documentation for DAML triggers

### PR DESCRIPTION
**Pull request by cocreature**
The API will still change in a bunch of ways but I’d rather get some
docs in place now and update them as we change things than not have
any docs at all.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/triggers/index.rst [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+initialize your user-defined state based on the active contract set.
+
+The ``updateState`` function is called on new transactions and command
+completions and can be used to update your user-defined state based on
+the ACS and the transaction or completion. Since our DAML trigger does
+not have any interesting user-defined state, we will not go into
+details here.
+
+Finally, the ``rule`` function is the core of a DAML trigger. It
+defines which commands need to be send to the ledger based on the
```
Comment (aherrmann-da) [inline docs/source/triggers/index.rst]:
```suggestion
defines which commands need to be sent to the ledger based on the
```

File: docs/source/triggers/index.rst [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+the ``subscribedTo`` field and the ``Copy`` contracts where we are the
+``owner`` of the corresponding ``Original``.
+
+The commands in flight will be useful to avoid sending the same
+command multiple times if ``copyRule`` is run multiple times before we
+get the corresponding transaction. Note that DAML triggers are
+expected to be designed such that they can cope with this, e.g., after
+a restart or a crash where the commands in flight do not contain
+commands in flight from before the restart, so this is an optimization
+rather than something required for them to function correctly.
```
Comment (aherrmann-da) [inline docs/source/triggers/index.rst]: I find this sentence hard to parse. Maybe something like

Note that DAML triggers are expected to be robust against missing entries in the commands in flight. In particular they are not repopulated after a crash or restart. So, this is an optimization and should not be required for a trigger to function correctly.

File: docs/source/triggers/index.rst [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+In addition to duplicate copies, we also need to archive copies where
+the corresponding ``Original`` or ``Subscriber`` no longer exists.
+
+.. literalinclude:: ./template-root/src/CopyTrigger.daml
+   :language: daml
+   :start-after: -- ARCHIVE_COPIES_BEGIN
+   :end-before: -- ARCHIVE_COPIES_END
+
+To send the corresponding archve commands to the ledger, we iterate
```
Comment (aherrmann-da) [inline docs/source/triggers/index.rst]:
```suggestion
To send the corresponding archive commands to the ledger, we iterate
```

File: docs/source/triggers/index.rst [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+.. literalinclude:: ./template-root/src/CopyTrigger.daml
+   :language: daml
+   :start-after: -- TO_CREATE_COPY_BEGIN
+   :end-before: -- TO_CREATE_COPY_END
+
+Equipped with ``toCreateCopy``, we can define the set of
+``pendingCopies`` by filtering the commands in flight and the set of
+``eventualCopies``, i.e., either pending copies or copies that already
+exist.  The copies that we need to create are then the set of
+``neeedCopies`` that are not already in ``pendingCopies``. Similar to
```
Comment (aherrmann-da) [inline docs/source/triggers/index.rst]:
```suggestion
``neededCopies`` that are not already in ``pendingCopies``. Similar to
```

File: docs/source/triggers/index.rst (hunk 2 above)
Comment (cocreature) [reply docs/source/triggers/index.rst]: This changed after I merged @hurryabit’s suggestions. Does the new formulation help?

Comment (aherrmann-da) [reply docs/source/triggers/index.rst]: Yes, I think it's easier to read in https://github.com/digital-asset/daml/pull/3183.
